credentials.go

```diff
@@ -153,6 +158,25 @@ Error: %w`, err)
 	return provider, creds.Source, err
 }
 
+func webIdentityCredentialsProvider(ctx context.Context, awsConfig aws.Config, c *Config) (aws.CredentialsProvider, error) {
+	ar := c.AssumeRole
+	client := stsClient(awsConfig, c)
+
+	appCreds := stscreds.NewWebIdentityRoleProvider(client, ar.RoleARN, c, func(opts *stscreds.WebIdentityRoleOptions) {
+		opts.RoleSessionName = ar.SessionName
```

Comment:
FYI `stscreds.WebIdentityRoleProvider` doesn't currently support the `Policy` field from `sts.AssumeRoleWithWebIdentityInput`. I've created https://github.com/aws/aws-sdk-go-v2/issues/1662 and may create a PR in the SDK